src/opencv.mk

```diff
@@ -39,10 +40,12 @@ define $(PKG)_BUILD
       -DBUILD_TIFF=OFF \
       -DBUILD_JASPER=OFF \
       -DBUILD_JPEG=OFF \
+      -DBUILD_WEBP=OFF \
```

Comment:
libwebp was added as a dependency but disabled here.

Reply:
Yes, that's intentional. As far as I can tell those flags control 3rdparty libraries for which we want to use opencv's bundled version instead of whatever's on the system. I figure the desired behavior is to use as few of those as possible, yes?

So, I only left protobuf (and correspondingly, removed it from dependencies) because mxe protobuf breaks the build. Since the webp version currently present on mxe works, I decided to use that. Understand?

Reply:
> instead of whatever's on the system

system=MXE, I guess

> I figure the desired behavior is to use as few of those as possible, yes?

The desired behavior is to use MXE's version of a library instead of bundled.

> mxe protobuf breaks the build

MXE uses protobuf 3.3.2, opencv is bundled with protobuf 3.1.0.

What is the error message if MXE's protobuf is used?

Reply:
I posted the entire log above; this is the actual error:

```
In file included from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:5:0:
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:17:2: error: #error This file was generated by an older version of protoc which is
 #error This file was generated by an older version of protoc which is
  ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:18:2: error: #error incompatible with your Protocol Buffer headers. Please
 #error incompatible with your Protocol Buffer headers.  Please
  ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:19:2: error: #error regenerate this file with a newer version of protoc.
 #error regenerate this file with a newer version of protoc.
  ^
In file included from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:32:0,
                 from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:5:
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/tensor.pb.h:17:2: error: #error This file was generated by an older version of protoc which is
 #error This file was generated by an older version of protoc which is
  ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/tensor.pb.h:18:2: error: #error incompatible with your Protocol Buffer headers. Please
 #error incompatible with your Protocol Buffer headers.  Please
  ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/tensor.pb.h:19:2: error: #error regenerate this file with a newer version of protoc.
 #error regenerate this file with a newer version of protoc.
  ^
In file included from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/tensor.pb.h:30:0,
                 from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:32,
                 from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:5:
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/tensor_shape.pb.h:17:2: error: #error This file was generated by an older version of protoc which is
 #error This file was generated by an older version of protoc which is
  ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/tensor_shape.pb.h:18:2: error: #error incompatible with your Protocol Buffer headers. Please
 #error incompatible with your Protocol Buffer headers.  Please
  ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/tensor_shape.pb.h:19:2: error: #error regenerate this file with a newer version of protoc.
 #error regenerate this file with a newer version of protoc.
  ^
In file included from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/tensor.pb.h:31:0,
                 from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:32,
                 from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:5:
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/types.pb.h:17:2: error: #error This file was generated by an older version of protoc which is
 #error This file was generated by an older version of protoc which is
  ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/types.pb.h:18:2: error: #error incompatible with your Protocol Buffer headers. Please
 #error incompatible with your Protocol Buffer headers.  Please
  ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/types.pb.h:19:2: error: #error regenerate this file with a newer version of protoc.
 #error regenerate this file with a newer version of protoc.
  ^
In file included from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/tensor.pb.h:30:0,
                 from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:32,
                 from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:5:
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/tensor_shape.pb.h: In member function 'const string& tensorflow::TensorShapeProto_Dim::name() const':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/tensor_shape.pb.h:299:80: error: no matching function for call to 'google::protobuf::internal::ArenaStringPtr::Get(const string*) const'
   return name_.Get(&::google::protobuf::internal::GetEmptyStringAlreadyInited());
                                                                                ^
In file included from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:23:0,
                 from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:5:
/opt/mxe/usr/i686-w64-mingw32.static/include/google/protobuf/arenastring.h:66:31: note: candidate: const string& google::protobuf::internal::ArenaStringPtr::Get() const
   inline const ::std::string& Get() const { return *ptr_; }
                               ^
/opt/mxe/usr/i686-w64-mingw32.static/include/google/protobuf/arenastring.h:66:31: note:   candidate expects 0 arguments, 1 provided
In file included from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:32:0,
                 from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:5:
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/tensor.pb.h: In member function 'const string& tensorflow::TensorProto::tensor_content() const':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/tensor.pb.h:404:90: error: no matching function for call to 'google::protobuf::internal::ArenaStringPtr::Get(const string*) const'
   return tensor_content_.Get(&::google::protobuf::internal::GetEmptyStringAlreadyInited());
                                                                                          ^
In file included from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:23:0,
                 from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:5:
/opt/mxe/usr/i686-w64-mingw32.static/include/google/protobuf/arenastring.h:66:31: note: candidate: const string& google::protobuf::internal::ArenaStringPtr::Get() const
   inline const ::std::string& Get() const { return *ptr_; }
                               ^
/opt/mxe/usr/i686-w64-mingw32.static/include/google/protobuf/arenastring.h:66:31: note:   candidate expects 0 arguments, 1 provided
In file included from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:5:0:
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h: At global scope:
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:634:9: error: type/value mismatch at argument 3 in template parameter list for 'template<class Derived, class Key, class T, google::protobuf::internal::WireFormatLite::FieldType key_wire_type, google::protobuf::internal::WireFormatLite::FieldType value_wire_type, int default_enum_value> class google::protobuf::internal::MapField'
       0 > attr_;
         ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:634:9: note:   expected a type, got 'TYPE_STRING'
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:634:9: error: invalid conversion from 'int' to 'google::protobuf::internal::WireFormatLite::FieldType' [-fpermissive]
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h: In member function 'const string& tensorflow::AttrValue::s() const':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:913:86: error: no matching function for call to 'google::protobuf::internal::ArenaStringPtr::Get(const string*) const'
     return value_.s_.Get(&::google::protobuf::internal::GetEmptyStringAlreadyInited());
                                                                                      ^
In file included from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:23:0,
                 from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:5:
/opt/mxe/usr/i686-w64-mingw32.static/include/google/protobuf/arenastring.h:66:31: note: candidate: const string& google::protobuf::internal::ArenaStringPtr::Get() const
   inline const ::std::string& Get() const { return *ptr_; }
                               ^
/opt/mxe/usr/i686-w64-mingw32.static/include/google/protobuf/arenastring.h:66:31: note:   candidate expects 0 arguments, 1 provided
In file included from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:5:0:
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h: In member function 'const string& tensorflow::AttrValue::placeholder() const':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:1490:96: error: no matching function for call to 'google::protobuf::internal::ArenaStringPtr::Get(const string*) const'
     return value_.placeholder_.Get(&::google::protobuf::internal::GetEmptyStringAlreadyInited());
                                                                                                ^
In file included from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:23:0,
                 from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:5:
/opt/mxe/usr/i686-w64-mingw32.static/include/google/protobuf/arenastring.h:66:31: note: candidate: const string& google::protobuf::internal::ArenaStringPtr::Get() const
   inline const ::std::string& Get() const { return *ptr_; }
                               ^
/opt/mxe/usr/i686-w64-mingw32.static/include/google/protobuf/arenastring.h:66:31: note:   candidate expects 0 arguments, 1 provided
In file included from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:5:0:
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h: In member function 'const string& tensorflow::NameAttrList::name() const':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:1604:80: error: no matching function for call to 'google::protobuf::internal::ArenaStringPtr::Get(const string*) const'
   return name_.Get(&::google::protobuf::internal::GetEmptyStringAlreadyInited());
                                                                                ^
In file included from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:23:0,
                 from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:5:
/opt/mxe/usr/i686-w64-mingw32.static/include/google/protobuf/arenastring.h:66:31: note: candidate: const string& google::protobuf::internal::ArenaStringPtr::Get() const
   inline const ::std::string& Get() const { return *ptr_; }
                               ^
/opt/mxe/usr/i686-w64-mingw32.static/include/google/protobuf/arenastring.h:66:31: note:   candidate expects 0 arguments, 1 provided
In file included from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:5:0:
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h: In member function 'int tensorflow::NameAttrList::attr_size() const':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:1666:16: error: request for member 'size' in '((const tensorflow::NameAttrList*)this)->tensorflow::NameAttrList::attr_', which is of non-class type 'const int'
   return attr_.size();
                ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h: In member function 'void tensorflow::NameAttrList::clear_attr()':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:1669:9: error: request for member 'Clear' in '((tensorflow::NameAttrList*)this)->tensorflow::NameAttrList::attr_', which is of non-class type 'int'
   attr_.Clear();
         ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h: In member function 'const google::protobuf::Map<std::__cxx11::basic_string<char>, tensorflow::AttrValue>& tensorflow::NameAttrList::attr() const':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:1674:16: error: request for member 'GetMap' in '((const tensorflow::NameAttrList*)this)->tensorflow::NameAttrList::attr_', which is of non-class type 'const int'
   return attr_.GetMap();
                ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h: In member function 'google::protobuf::Map<std::__cxx11::basic_string<char>, tensorflow::AttrValue>* tensorflow::NameAttrList::mutable_attr()':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:1679:16: error: request for member 'MutableMap' in '((tensorflow::NameAttrList*)this)->tensorflow::NameAttrList::attr_', which is of non-class type 'int'
   return attr_.MutableMap();
                ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc: In function 'void tensorflow::protobuf_AssignDesc_attr_5fvalue_2eproto()':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:72:5: error: 'NewGeneratedMessageReflection' is not a member of 'google::protobuf::internal::GeneratedMessageReflection'
     ::google::protobuf::internal::GeneratedMessageReflection::NewGeneratedMessageReflection(
     ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:94:5: error: 'NewGeneratedMessageReflection' is not a member of 'google::protobuf::internal::GeneratedMessageReflection'
     ::google::protobuf::internal::GeneratedMessageReflection::NewGeneratedMessageReflection(
     ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:109:5: error: 'NewGeneratedMessageReflection' is not a member of 'google::protobuf::internal::GeneratedMessageReflection'
     ::google::protobuf::internal::GeneratedMessageReflection::NewGeneratedMessageReflection(
     ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc: In function 'void tensorflow::{anonymous}::protobuf_RegisterTypes(const string&)':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:145:14: error: wrong number of template arguments (5, should be 6)
             0>::CreateDefaultInstance(
              ^
In file included from /opt/mxe/usr/i686-w64-mingw32.static/include/google/protobuf/map_entry.h:35:0,
                 from /opt/mxe/usr/i686-w64-mingw32.static/include/google/protobuf/map_field.h:40,
                 from /opt/mxe/usr/i686-w64-mingw32.static/include/google/protobuf/map_field_inl.h:40,
                 from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:30,
                 from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:5:
/opt/mxe/usr/i686-w64-mingw32.static/include/google/protobuf/map_entry_lite.h:46:7: note: provided for 'template<class Derived, class Key, class Value, google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType, google::protobuf::internal::WireFormatLite::FieldType kValueFieldType, int default_enum_value> class google::protobuf::internal::MapEntry'
 class MapEntry;
       ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc: In function 'void tensorflow::{anonymous}::MergeFromFail(int)':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:233:3: error: 'MergeFromFail' is not a member of 'google::protobuf::internal'
   ::google::protobuf::internal::MergeFromFail(__FILE__, line);
   ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:233:3: note: suggested alternatives:
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:232:13: note:   'tensorflow::{anonymous}::MergeFromFail'
 static void MergeFromFail(int line) {
             ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:232:13: note:   'tensorflow::{anonymous}::MergeFromFail'
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc: In member function 'void tensorflow::AttrValue_ListValue::UnsafeMergeFrom(const tensorflow::AttrValue_ListValue&)':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:787:6: error: 'class google::protobuf::RepeatedPtrField<std::__cxx11::basic_string<char> >' has no member named 'UnsafeMergeFrom'
   s_.UnsafeMergeFrom(from.s_);
      ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:788:6: error: 'class google::protobuf::RepeatedField<long long int>' has no member named 'UnsafeMergeFrom'
   i_.UnsafeMergeFrom(from.i_);
      ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:789:6: error: 'class google::protobuf::RepeatedField<float>' has no member named 'UnsafeMergeFrom'
   f_.UnsafeMergeFrom(from.f_);
      ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:790:6: error: 'class google::protobuf::RepeatedField<bool>' has no member named 'UnsafeMergeFrom'
   b_.UnsafeMergeFrom(from.b_);
      ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:791:9: error: 'class google::protobuf::RepeatedField<int>' has no member named 'UnsafeMergeFrom'
   type_.UnsafeMergeFrom(from.type_);
         ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc: In constructor 'tensorflow::NameAttrList::NameAttrList(google::protobuf::Arena*)':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2499:14: error: invalid conversion from 'google::protobuf::Arena*' to 'int' [-fpermissive]
   attr_(arena) {
              ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc: In member function 'void tensorflow::NameAttrList::SharedCtor()':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2520:9: error: request for member 'SetAssignDescriptorCallback' in '((tensorflow::NameAttrList*)this)->tensorflow::NameAttrList::attr_', which is of non-class type 'int'
   attr_.SetAssignDescriptorCallback(
         ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2522:9: error: request for member 'SetEntryDescriptor' in '((tensorflow::NameAttrList*)this)->tensorflow::NameAttrList::attr_', which is of non-class type 'int'
   attr_.SetEntryDescriptor(
         ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc: In member function 'virtual void tensorflow::NameAttrList::Clear()':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2572:9: error: request for member 'Clear' in '((tensorflow::NameAttrList*)this)->tensorflow::NameAttrList::attr_', which is of non-class type 'int'
   attr_.Clear();
         ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc: In member function 'virtual bool tensorflow::NameAttrList::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*)':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2611:17: error: type/value mismatch at argument 3 in template parameter list for 'template<class Derived, class Key, class T, google::protobuf::internal::WireFormatLite::FieldType key_wire_type, google::protobuf::internal::WireFormatLite::FieldType value_wire_type, int default_enum_value> class google::protobuf::internal::MapField'
               0 >,
                 ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2611:17: note:   expected a type, got 'TYPE_STRING'
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2611:17: error: invalid conversion from 'int' to 'google::protobuf::internal::WireFormatLite::FieldType' [-fpermissive]
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2612:79: error: template argument 1 is invalid
             ::google::protobuf::Map< ::std::string, ::tensorflow::AttrValue > > parser(&attr_);
                                                                               ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2612:94: error: invalid conversion from 'int*' to 'int' [-fpermissive]
             ::google::protobuf::Map< ::std::string, ::tensorflow::AttrValue > > parser(&attr_);
                                                                                              ^
In file included from /opt/mxe/usr/i686-w64-mingw32.static/include/google/protobuf/stubs/common.h:40:0,
                 from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:9,
                 from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:5:
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2616:20: error: request for member 'key' in 'parser', which is of non-class type 'int'
             parser.key().data(), parser.key().length(),
                    ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2615:11: note: in expansion of macro 'DO_'
           DO_(::google::protobuf::internal::WireFormatLite::VerifyUtf8String(
           ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2616:41: error: request for member 'key' in 'parser', which is of non-class type 'int'
             parser.key().data(), parser.key().length(),
                                         ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2615:11: note: in expansion of macro 'DO_'
           DO_(::google::protobuf::internal::WireFormatLite::VerifyUtf8String(
           ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc: In member function 'virtual void tensorflow::NameAttrList::SerializeWithCachedSizes(google::protobuf::io::CodedOutputStream*) const':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2677:17: error: 'class google::protobuf::io::CodedOutputStream' has no member named 'IsSerializationDeterminstic'
     if (output->IsSerializationDeterminstic() &&
                 ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2691:27: error: request for member 'NewEntryWrapper' in '((const tensorflow::NameAttrList*)this)->tensorflow::NameAttrList::attr_', which is of non-class type 'const int'
         entry.reset(attr_.NewEntryWrapper(
                           ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2705:27: error: request for member 'NewEntryWrapper' in '((const tensorflow::NameAttrList*)this)->tensorflow::NameAttrList::attr_', which is of non-class type 'const int'
         entry.reset(attr_.NewEntryWrapper(
                           ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc: In member function 'virtual google::protobuf::uint8* tensorflow::NameAttrList::InternalSerializeWithCachedSizesToArray(bool, google::protobuf::uint8*) const':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2764:27: error: request for member 'NewEntryWrapper' in '((const tensorflow::NameAttrList*)this)->tensorflow::NameAttrList::attr_', which is of non-class type 'const int'
         entry.reset(attr_.NewEntryWrapper(
                           ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2780:27: error: request for member 'NewEntryWrapper' in '((const tensorflow::NameAttrList*)this)->tensorflow::NameAttrList::attr_', which is of non-class type 'const int'
         entry.reset(attr_.NewEntryWrapper(
                           ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc: In member function 'virtual size_t tensorflow::NameAttrList::ByteSizeLong() const':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2820:25: error: request for member 'NewEntryWrapper' in '((const tensorflow::NameAttrList*)this)->tensorflow::NameAttrList::attr_', which is of non-class type 'const int'
       entry.reset(attr_.NewEntryWrapper(it->first, it->second));
                         ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc: In member function 'void tensorflow::NameAttrList::UnsafeMergeFrom(const tensorflow::NameAttrList&)':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2862:9: error: request for member 'MergeFrom' in '((tensorflow::NameAttrList*)this)->tensorflow::NameAttrList::attr_', which is of non-class type 'int'
   attr_.MergeFrom(from.attr_);
         ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc: In member function 'void tensorflow::NameAttrList::InternalSwap(tensorflow::NameAttrList*)':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2905:9: error: request for member 'Swap' in '((tensorflow::NameAttrList*)this)->tensorflow::NameAttrList::attr_', which is of non-class type 'int'
   attr_.Swap(&other->attr_);
         ^
In file included from /opt/mxe/usr/i686-w64-mingw32.static/include/google/protobuf/map_type_handler.h:36:0,
                 from /opt/mxe/usr/i686-w64-mingw32.static/include/google/protobuf/map.h:48,
                 from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.h:29,
                 from /opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:5:
/opt/mxe/usr/i686-w64-mingw32.static/include/google/protobuf/wire_format_lite_inl.h: In instantiation of 'static bool google::protobuf::internal::WireFormatLite::ReadMessageNoVirtual(google::protobuf::io::CodedInputStream*, MessageType_WorkAroundCppLookupDefect*) [with MessageType = int]':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:2613:11:   required from here
/opt/mxe/usr/i686-w64-mingw32.static/include/google/protobuf/wire_format_lite_inl.h:494:23: error: request for member 'int:: MergePartialFromCodedStream' in '* value', which is of non-class type 'int'
   if (p.second < 0 || !value->
                       ^
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc: In function 'void tensorflow::{anonymous}::MergeFromFail(int)':
/opt/mxe/tmp-opencv-i686-w64-mingw32.static/opencv-3.3.0/modules/dnn/misc/tensorflow/attr_value.pb.cc:234:1: warning: 'noreturn' function does return
 }
 ^
make[4]: *** [modules/dnn/CMakeFiles/opencv_dnn.dir/misc/tensorflow/attr_value.pb.cc.obj] Error 1
make[3]: *** [modules/dnn/CMakeFiles/opencv_dnn.dir/all] Error 2
make[3]: *** Waiting for unfinished jobs....
```

Reply:
opencv has an option `PROTOBUF_UPDATE_FILES` to rebuild generated protobuf files.
Does it build with this option?

Reply:
Yes; that works. I've updated the file.